utils/utf8.hh

```diff
@@ -64,6 +65,54 @@ inline std::optional<size_t> validate_with_error_position(bytes_view string) {
     return validate_with_error_position(data, len);	
 }
 
+std::optional<size_t> validate_with_error_position_fragmented(const FragmentRange auto& fr) {
+    uint8_t partial_codepoint[4];
```

Comment:
std::array?

Reply:
why?

Reply:
better safety and more idiomatic for modern c++